### PR TITLE
Sequence diagrams done

### PR DESCRIPTION
**Pull request by danutz87**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,33 +21,46 @@ const port = process.env.PORT;
 async function createApp() {
   await connectToDb();
 
+  app.use(
+    expressWinston.logger({
+      transports: [new winston.transports.Console()],
```
Comment (bogdanned) [inline index.js]: Should depend on the .NODE_ENV variable

File: index.js [hunk 2]
```diff
@@ -21,33 +21,46 @@ const port = process.env.PORT;
...
+        winston.format.json()
+      ),
+    })
+  );
+
   const app = express();
   // Middleware: parse the body to json
   app.use(bodyParser.json());
   // Api documentation
   app.use("/api-docs", swaggerUi.serve, swaggerUi.setup(swaggerDocument));
```
Comment (bogdanned) [inline index.js]: To-do: Create a Winston logger and add it to the res object, so it can be used later to log

File: index.js [hunk 3]
```diff
@@ -21,33 +21,46 @@ const port = process.env.PORT;
...
     } catch (err) {
-      res.status(500).json({ message: err.message });
+      res.status(500).json({ message: "Internal server error" });
     }
   });
   // Retrieve an order with the orderId
   app.get("/:orderId", async (req, res) => {
     const { orderId } = req.params;
+
+    if (!orderId) res.status(400).end();
```
Comment (bogdanned) [inline index.js]: Can also send and error message with `res.json({message: "Order not found"})`

File: index.js [hunk 4]
```diff
@@ -21,33 +21,46 @@ const port = process.env.PORT;
...
     try {
       const order = await Order.findOne({ id: orderId });
       if (!order) {
```
Comment (bogdanned) [inline index.js:52]: `if (order) {`

File: index.js [hunk 5]
```diff
@@ -56,7 +69,7 @@ async function createApp() {
         res.send(order);
       }
```
Comment (bogdanned) [inline index.js:56]:
```js
else{
     res.status(404).json({message: "order not found"})
}
```

File: index.js [hunk 6]
```diff
@@ -70,17 +83,19 @@ async function createApp() {
       const newOrder = await order.save();
       res.status(201).json(newOrder);
     } catch (err) {
-      res.status(400).json({ message: err.message });
+      res.status(400).json({ message: "Bad request" });
```
Comment (bogdanned) [inline index.js]: 500 as is an error message

File: index.js [hunk 7]
```diff
@@ -70,17 +83,19 @@ async function createApp() {
...
     }
   });
 
   app.delete("/:orderId", async (req, res) => {
     const { orderId } = req.params;
+
+    if (!orderId) res.status(400).end();
     try {
       await Order.findOneAndDelete({ id: orderId });
       res.json({ message: "The order was deleted" });
```
Comment (bogdanned) [inline index.js]: status(204)

**Comment by bogdanned** [comment]
@danutz87 I left some comments mainly about status codes

**Comment by bogdanned** [comment]
# TODO:
1. Extra: Add nodemon --> DONE
2. Extra:Add typescript
4. Deploy and a live link in the README
5. Move dotenv to be a "development" only dependency
6. Set up prettier and a linter

**Review by bogdanned — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (danutz87) [reply index.js]: Done

**Review by bogdanned — APPROVED** [review]

**Pull request merged**